AitGroup25Assignment/src/ai2018/group25/ConstBid.java

```diff
@@ -0,0 +1,170 @@
+package ai2018.group25;
+
+/* IMPORTANT, JUST IGNORE ALL, OF THIS, THIS IS THE EXAMPLE GIVEN, WE USED IT BECAUSE IT HAS THE STRUCTURE
+ * WORKING, BUT WE JUST COMMENTED ALL THE LOGIC OF determineNextBid() AND MADE THIS EXAMPLE GIVE A 
+ * CONSTANT BID, JUST TO TEST THE ACCEPTANCE STRATEGY PROPERLY IN AN EASY WAY
+ */
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import genius.core.bidding.BidDetails;
+import genius.core.boaframework.BOAparameter;
+import genius.core.boaframework.NegotiationSession;
+//import genius.core.boaframework.NoModel;
+import genius.core.boaframework.OMStrategy;
+import genius.core.boaframework.OfferingStrategy;
+import genius.core.boaframework.OpponentModel;
+import genius.core.boaframework.SortedOutcomeSpace;
+
+/**
+ * This is an abstract class used to implement a TimeDependentAgent Strategy
+ * adapted from [1] [1] S. Shaheen Fatima Michael Wooldridge Nicholas R.
+ * Jennings Optimal Negotiation Strategies for Agents with Incomplete
+ * Information http://eprints.ecs.soton.ac.uk/6151/1/atal01.pdf
+ * 
+ * The default strategy was extended to enable the usage of opponent models.
+ * 
+ * Note that this agent is not fully equivalent to the theoretical model,
+ * loading the domain may take some time, which may lead to the agent skipping
+ * the first bid. A better implementation is GeniusTimeDependent_Offering.
+ */
+@SuppressWarnings("deprecation")
+public class ConstBid extends OfferingStrategy {
+
+	/**
+	 * k in [0, 1]. For k = 0 the agent starts with a bid of maximum utility
+	 */
+	private double k;
+	/** Maximum target utility */
+	private double Pmax;
+	/** Minimum target utility */
+	private double Pmin;
+	/** Concession factor */
+	private double e;
```

Comment:
I would go for
`private double concessionFactor` but that is upto you

Reply:
I justed used what was done and commented it all as you should have seen...